<a href="https://colab.research.google.com/github/jesusvasquez-tec/analitica-empresarial-integrada/blob/main/Semana2/Vasquez_LAB_D2_AEI_PSAYAN_2026_02.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# LABORATORIO DIRIGIDO N.° 02
## Analítica Empresarial Integrada — TECSUP 2026-II

| | |
|---|---|
| **Docente** | Pilar Rocío Sayán Mejía |
| **Laboratorio** | Perfil analítico comparado de empresas del sector azucarero peruano |
| **N.° de laboratorio** | 2 |
| **Tema** | Estadios de evolución analítica y modelo DELTA de madurez organizacional |
| **Plan curricular** | Big Data y Ciencia de Datos · Código C28 · 2026-II |
| **Duración** | 190 minutos (7:00 p.m. – 10:10 p.m., con receso de 15 minutos) |
| **Fuente de datos** | SMV y BCRP — datos reales y oficiales |
| **Modalidad** |

---

**Equipo N.°:** 4 &nbsp;&nbsp;&nbsp; **Sección:** 6C28A &nbsp;&nbsp;&nbsp; **Fecha:** 26-08-2026

**Apellidos y nombres del estudiante:** Vasquez Espinoza, Jesus

**Integrantes del equipo:**  

* Alvino Carhuaricra, Enzo
* Cordova Agurto, Isabel
* Palomino Acuña, Luis
* Vasquez Espinoza, Jesus



---


**Elemento de la capacidad terminal:** comprende los estadios de evolución analítica y las dimensiones del modelo DELTA y de su ampliación DELTA Plus; aplica estos marcos a la comparación de indicadores financieros reales y diferencia evidencia, inferencia y opinión.

> **Alcance.** Se analizan estados financieros individuales y anuales del ejercicio 2024. Los importes de la SMV se expresan en miles y pueden estar reportados en soles o dólares según la empresa. Las razones financieras son comparables cuando el numerador y el denominador pertenecen a la misma empresa, moneda y periodo.


## Organización de la sesión — 7:00 p. m. a 10:10 p. m.

**Duración total:** 190 minutos · **Receso:** 15 minutos · **Trabajo efectivo:** 175 minutos.

| Horario | Tiempo | Desarrollo |
|---|---:|---|
| 7:00–7:10 | 10 min | Apertura del caso y conexión con los estadios de evolución analítica. |
| 7:10–7:35 | 25 min | Actividad 1. Revisión de conceptos. Modelos DELTA y DELTA Plus. |
| 7:35–8:00 | 25 min | Actividad 2. Entorno y descarga de información financiera de la SMV. |
| 8:00–8:30 | 30 min | Actividad 2. Normalización, validación e indicadores financieros. Reto 1. |
| 8:30–8:45 | 15 min | **RECESO** |
| 8:45–9:15 | 30 min | Actividad 2. Comparación con Polars y DuckDB. Visualización. Reto 2. |
| 9:15–9:35 | 20 min | Actividad 2. Contexto sectorial del BCRP. Reto 3. |
| 9:35–10:00 | 25 min | **Reto de aplicación y retroalimentación.** Ejercicios 1 a 5. |
| 10:00–10:10 | 10 min | Actividad 3. Informe, conclusiones y preguntas de DELTA Plus. |

> **Regla de trabajo:** no avance de bloque sin registrar la interpretación solicitada. El objetivo no es ejecutar celdas, sino convertir datos en evidencia para una decisión.


## Actividad 1 — Revisión de conceptos (25 minutos)

**Propósito.** Consolidar el vocabulario técnico de la semana antes del desarrollo práctico. La precisión conceptual constituye un requisito del análisis: un indicador correctamente calculado pierde valor si se interpreta con categorías imprecisas.

**Instrucciones.** Complete la tabla con definiciones elaboradas con sus propias palabras. No se admite la reproducción literal de fuentes externas ni de sistemas generativos. La columna de la derecha contiene una pregunta de apoyo: si su definición permite responderla, la definición es suficiente; si no lo permite, corríjala antes de continuar.

**Evidencia esperada.** Tabla completa con las definiciones registradas.

| Concepto | Definición elaborada por el estudiante | Pregunta de apoyo |
|---|---|---|
| Estadios de la evolución analítica | Son niveles que muestran cómo una organización pasa de usar los datos de manera básica o<br> aislada a utilizarlos de forma estratégica para tomar decisiones y competir | ¿Qué distingue a una organización del estadio 2 de una del estadio 3? |
| Modelo DELTA |  (Datos, <br>Empresa, Liderazgo, Objetivos y Analistas) Es un modelo que evalúa la capacidad analítica de <br>una organización considerando varias dimensiones que deben desarrollarse de manera coordinada. | ¿Por qué el modelo sostiene que las dimensiones deben avanzar de manera conjunta? |
| Modelo DELTA Plus | Ampliación del modelo DELTA que incorpora las dimensiones de Tecnología y Técnicas Analíticas<br> para corregir la limitación del modelo original, reconociendo que contar con datos y talento no es <br>suficiente si la organización carece de la infraestructura técnica o de metodologías avanzadas para <br>ejecutar el análisis. | ¿Qué limitación del modelo original corrige la ampliación? |
| Dimensión Technology | Componente del modelo DELTA Plus que evalúa la infraestructura y plataformas necesarias para <br>procesar datos, justificando que una organización puede tener madurez limitada si, aun contando <br>con datos y analistas, carece de la tecnología para operar de forma periódica y reproducible. | ¿Puede una organización con datos y analistas presentar madurez limitada? Justifique. |
| Dimensión Analytical Techniques | Dimensión que mide la sofisticación de los métodos usados para convertir datos en información, <br>abarcando desde la analítica descriptiva (que indica qué ocurrió, como calcular un margen) hasta<br> la diagnóstica (que indaga por qué ocurrió, como analizar las causas de su caída). | ¿Qué diferencia una técnica descriptiva de una diagnóstica? Proponga un ejemplo. |
| Organización que compite con analítica | Empresa donde la analítica define la estrategia central y crea ventajas sostenibles, lo cual requiere<br> no solo tableros de control, sino un liderazgo que exija evidencia, métricas compartidas y un talento <br>analítico directamente conectado con la toma de decisiones reales. | ¿Basta con disponer de tableros de control para competir con analítica? |
| Ventaja competitiva basada en datos | Es el beneficio que obtiene una empresa cuando transforma sus datos, capacidades análiticas, <br>experiencia y procesos en mejores decisiones que sus competidores | ¿Por qué razones esta ventaja puede resultar difícil de imitar por un competidor? |
| Impacto de la IA generativa en la empresa | Es el cambio que produce la IA generativa al apoyar y acelerar tareas como resumir, explorar <br>información, generar alternativas o automatizar procesos, sin reemplazar el criterio profesional<br> en decisiones críticas. | ¿Qué tarea del análisis financiero no debería delegarse a un sistema generativo? |
| Perfil analítico comparado | Para que dos empresas sean comparables, deberían tener características suficientemente similares, <br> principalmente: Pertenecer al mismo sector o actividad económica, Tener un modelo de negocio <br>similar, Compararse en el mismo período de tiempo, Utilizar indicadores calculados con los mismos <br>criterios, Considerar diferencias importantes de tamaño, mercado y contexto | ¿Qué condiciones deben cumplirse para que dos empresas sean comparables? |
| Indicador financiero como evidencia | Dato cuantitativo derivado de los estados financieros que respalda conclusiones sobre el desempeño <br>económico, pero que no constituye evidencia válida si se utiliza para inferir capacidades organizacionales <br>no financieras, como la madurez analítica o la cultura de datos. | ¿En qué caso un indicador correctamente calculado no constituye evidencia válida? |

**Criterio de cierre.** No se avanza al desarrollo práctico mientras existan conceptos sin definición registrada.


## Marco conceptual — Del modelo DELTA al modelo DELTA Plus

El modelo DELTA fue formulado por Davenport, Harris y Morison en *Analytics at Work* (2010) como un instrumento para evaluar la madurez analítica de una organización a partir de cinco dimensiones: **Data**, referida a la disponibilidad, integración y calidad de los datos; **Enterprise**, que examina si la capacidad analítica se encuentra dispersa en áreas aisladas o coordinada a escala institucional; **Leadership**, que valora el compromiso efectivo de la dirección con la decisión basada en evidencia; **Targets**, que analiza si los esfuerzos analíticos se concentran en los objetivos de mayor valor para el negocio; y **Analysts**, que considera la disponibilidad de profesionales con las competencias requeridas. El principio central del modelo establece que estas dimensiones deben progresar de manera conjunta, dado que un avance aislado en una de ellas no eleva la madurez global de la organización.

Con posterioridad, y ante la transformación del entorno tecnológico de la analítica empresarial, el modelo fue ampliado bajo la denominación **DELTA Plus**, que incorpora dos dimensiones adicionales. La primera es **Technology**, que comprende la infraestructura y las plataformas que permiten desarrollar y desplegar capacidades analíticas: soluciones en la nube, *data warehouses*, *data lakes* y arquitecturas *lakehouse*, entornos de procesamiento de grandes volúmenes de información, herramientas de visualización, plataformas analíticas corporativas, componentes de código abierto e infraestructura para el desarrollo de modelos de aprendizaje automático e inteligencia artificial. Su incorporación obedece a una constatación de orden práctico: una organización puede disponer de datos pertinentes y de analistas calificados y presentar, aun así, una madurez analítica limitada si carece de la tecnología necesaria para almacenar, procesar, analizar y poner en uso esa información de manera periódica y reproducible.

La segunda dimensión añadida es **Analytical Techniques**, que evalúa la variedad y la sofisticación de las técnicas empleadas para convertir los datos en información útil para la decisión. Estas técnicas admiten una progresión reconocible: la analítica **descriptiva** establece qué ocurrió; la **diagnóstica** indaga por qué ocurrió; la **predictiva** estima qué podría ocurrir; y la **prescriptiva** determina qué conviene hacer. Sobre esa base se incorporan progresivamente el aprendizaje automático, las técnicas de segmentación, los modelos de clasificación, los pronósticos, los métodos de optimización y las aplicaciones de inteligencia artificial. Dichas técnicas se presentan aquí como ilustración de la evolución de la capacidad analítica de una organización y serán objeto de desarrollo en semanas posteriores del curso; el presente laboratorio se sitúa en los niveles descriptivo y diagnóstico.

### Tabla comparativa

| | Modelo DELTA (2010) | Modelo DELTA Plus |
|---|---|---|
| **Dimensiones** | Cinco | Siete |
| **Componentes** | Data, Enterprise, Leadership, Targets, Analysts | Las cinco anteriores, más Technology y Analytical Techniques |
| **Objeto de evaluación** | Condiciones organizacionales de la analítica | Condiciones organizacionales y capacidad técnica de ejecución |
| **Pregunta que responde** | ¿Dispone la organización de datos, dirección, objetivos y personas? | Además de lo anterior, ¿cuenta con la tecnología y las técnicas para ejecutarlo de manera sostenida? |
| **Limitación que corrige** | — | Una organización con datos y analistas puede presentar madurez limitada por ausencia de tecnología o por el uso de técnicas elementales |

> **Precisión conceptual.** DELTA y DELTA Plus no constituyen modelos alternativos. DELTA Plus es una ampliación del modelo original: conserva íntegramente sus cinco dimensiones y añade dos. Al emitir un diagnóstico debe indicarse explícitamente cuál de los dos marcos se está aplicando.


## Actividad 2 — Desarrollo práctico y ejecución

**Propósito:** descargar información financiera anual y real de empresas del sector azucarero peruano, construir un perfil comparado de rentabilidad, liquidez y endeudamiento, y relacionarlo prudentemente con el contexto sectorial del BCRP mediante pandas, Polars, DuckDB y Plotly.

**Empresas:** Casa Grande S.A.A., Cartavio S.A.A., Agroindustrial Laredo S.A.A. y Empresa Agroindustrial Pomalca S.A.A.


## Paso 1: Preparación del entorno


In [ ]:
# Ejecute esta celda en Google Colab. Las versiones se fijan para favorecer la reproducibilidad.
%pip install -q polars==1.17.1 duckdb==1.1.3 econdata==1.0.17

import json, html, re, unicodedata
from xml.etree import ElementTree as ET

import requests
import pandas as pd
import numpy as np
import polars as pl
import duckdb
import plotly.graph_objects as go
from IPython.display import display
from econdata import BCRP

pd.set_option("display.width", 160)
pd.set_option("display.max_columns", 40)
print("Entorno listo.")


   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 36.4/36.4 MB 19.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 20.2/20.2 MB 24.5 MB/s eta 0:00:00
Entorno listo.


## Paso 2: Descarga de información financiera real desde la SMV

Se emplea el servicio web oficial de la SMV para consultar las principales cuentas y el Estado de Situación Financiera. Las operaciones utilizadas son `obtener_InfoFinanciera` y `obtener_BalanceGeneral`. El nombre oficial `obtener_GanciaPerdida` también se conserva para consultas del Estado de Resultados.


In [ ]:
SERVICIO_SMV = "https://mvnet.smv.gob.pe/ws_od_eeff/WebServiceInfoFinanciera.asmx"
EJERCICIO = 2024
PERIODO = "A"   # Anual
TIPO = "I"      # Individual
AZUCARERAS = ["CASA GRANDE", "CARTAVIO", "AGROINDUSTRIAL LAREDO", "POMALCA"]

def etiqueta_local(etiqueta):
    '''Retira el espacio de nombres de una etiqueta XML.'''
    return str(etiqueta).split("}")[-1]

def payload_a_dataframe(payload):
    '''Convierte el contenido JSON o XML incluido en la respuesta SOAP en DataFrame.'''
    payload = html.unescape(str(payload)).strip()
    try:
        objeto = json.loads(payload)
        if isinstance(objeto, dict):
            objeto = objeto.get("rows", objeto.get("data", objeto))
        if isinstance(objeto, dict):
            objeto = [objeto]
        return pd.DataFrame(objeto)
    except json.JSONDecodeError:
        raiz = ET.fromstring(payload)
        registros = []
        for nodo in raiz.iter():
            hijos = list(nodo)
            if len(hijos) >= 5 and all(not list(hijo) for hijo in hijos):
                registros.append({etiqueta_local(h.tag): h.text for h in hijos})
        return pd.DataFrame(registros)

def descargar_smv(operacion, ejercicio=EJERCICIO, periodo=PERIODO, tipo=TIPO):
    '''Consulta una operación oficial del servicio SOAP de la SMV.'''
    sobre = f'''<?xml version="1.0" encoding="utf-8"?>
    <soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
      <soap:Body><{operacion} xmlns="http://tempuri.org/">
        <Ejercicio>{ejercicio}</Ejercicio><Periodo>{periodo}</Periodo><Tipo>{tipo}</Tipo>
      </{operacion}></soap:Body>
    </soap:Envelope>'''
    respuesta = requests.post(
        SERVICIO_SMV,
        data=sobre.encode("utf-8"),
        timeout=90,
        headers={
            "Content-Type": "text/xml; charset=utf-8",
            "SOAPAction": f'"http://tempuri.org/{operacion}"',
        },
    )
    respuesta.raise_for_status()
    raiz = ET.fromstring(respuesta.content)
    resultado = next(
        (n for n in raiz.iter() if etiqueta_local(n.tag) == f"{operacion}Result"),
        None,
    )
    if resultado is None or not (resultado.text or "").strip():
        raise ValueError(f"La SMV no devolvió datos para {operacion}.")
    datos = payload_a_dataframe(resultado.text)
    if datos.empty:
        raise ValueError(f"La respuesta de {operacion} no contiene registros.")
    return datos

principales = descargar_smv("obtener_InfoFinanciera")
balance = descargar_smv("obtener_BalanceGeneral")

print("Principales cuentas:", principales.shape)
print("Estado de situación financiera:", balance.shape)
display(principales.head(3))


Principales cuentas: (276, 16)
Estado de situación financiera: (20574, 15)


,RPJ,TipoEmpresa,TipoSector,NombreEmpresa,RUC,CIIU,Ejercicio,TipoInformacion,Trimestre,Moneda,MetodoFlujoEfectivo,ActivoTotal,PatrimonioTotal,TotalIngreso,UtilidadNeta,PasivoTotal
0,L00474,EMPRESAS MERCADO ALTERNATIVO DE VALORES,DIVERSOS,A. JAIME ROJAS REPRESENTACIONES GENERALES S.A.,20102032951,5190,2024,Anual Individual,Anual,Soles,M todo Directo,77710,39497,77196,6786,38213
1,I00004,SOCIEDADES ADMINISTRADORAS DE FONDOS DE INVERSION,,AC CAPITALES SOCIEDAD ADMINISTRADORA DE FONDOS...,20504893295,6430,2024,Anual Individual,Anual,Soles,M todo Directo,9070,7373,4880,-85,1697
2,OE7511,SOCIEDADES ADMINISTRADORAS DE FONDOS DE INVERSION,,ACRES SOCIEDAD ADMINISTRADORA DE FONDOS DE INV...,20601498996,6430,2024,Anual Individual,Anual,Soles,M todo Indirecto,2724,2238,2348,176,486


## Paso 3: Normalización, filtrado y validación del sector

Se normalizan los nombres de las columnas, se filtran las cuatro empresas y se verifica que todas estén representadas. No se utiliza una base simulada ni un respaldo inventado.


In [ ]:
def nombre_columna(texto):
    texto = re.sub(r"(?<!^)(?=[A-Z])", "_", str(texto))
    texto = unicodedata.normalize("NFKD", texto).encode("ascii", "ignore").decode()
    return re.sub(r"[^a-z0-9]+", "_", texto.lower()).strip("_")

def preparar(df):
    df = df.copy()
    df.columns = [nombre_columna(c) for c in df.columns]
    return df

def resolver_columna(df, candidatos):
    '''Encuentra una columna aun cuando la SMV cambie mayúsculas o convenciones de nombre.'''
    disponibles = {nombre_columna(c): c for c in df.columns}
    for candidato in candidatos:
        clave = nombre_columna(candidato)
        if clave in disponibles:
            return disponibles[clave]
    raise KeyError(f"No se encontró ninguna de estas columnas: {candidatos}. Disponibles: {list(df.columns)}")

def estandarizar_empresa(df):
    df = preparar(df)
    origen = resolver_columna(df, ["nombre_empresa", "razon_social", "empresa"])
    return df.rename(columns={origen: "nombre_empresa"})

def filtrar_azucareras(df):
    patron = "|".join(map(re.escape, AZUCARERAS))
    return df[
        df["nombre_empresa"].astype(str).str.upper().str.contains(patron, regex=True, na=False)
    ].copy()

principales = filtrar_azucareras(estandarizar_empresa(principales))
balance = filtrar_azucareras(estandarizar_empresa(balance))

empresas = sorted(principales["nombre_empresa"].dropna().unique())
print("Empresas encontradas:")
for empresa in empresas:
    print("-", empresa)

faltantes = [nombre for nombre in AZUCARERAS if not any(nombre in e.upper() for e in empresas)]
if faltantes:
    raise ValueError(f"No se localizaron todas las empresas. Faltantes: {faltantes}")


Empresas encontradas:
- AGROINDUSTRIAL LAREDO S.A.A.
- CARTAVIO SOCIEDAD ANONIMA ABIERTA
- CASA GRANDE SOCIEDAD ANONIMA ABIERTA
- EMPRESA AGROINDUSTRIAL POMALCA S.A.A.


### Paso 3: La cuenta correcta no se busca por su nombre

El Estado de Situación Financiera trae 69 cuentas por empresa, y **más de una contiene el texto «Activos Corrientes»**, pero solo una es el total. Filtrar por la descripción es un error frecuente y silencioso: no lanza ninguna excepción, devuelve un número, y ese número está mal.

Ejecute la celda siguiente y observe el catálogo: la cuenta `1D0122` —«Activos Corrientes Distintos al Efectivo Pignorados como Garantía Colateral»— vale 0 y, de usarse, produce una razón corriente igual a cero.


In [ ]:
# Catálogo de cuentas cuyo texto contiene "corriente", para una de las empresas.
muestra = balance[balance["nombre_empresa"] == empresas[0]]
columna_desc = resolver_columna(muestra, ["descripcion_cuenta", "descripcion"])
catalogo = muestra[
    muestra[columna_desc].astype(str).str.contains("orriente", na=False)
][["cuenta", columna_desc, "monto1"]]
print("Empresa de referencia:", empresas[0])
display(catalogo.reset_index(drop=True))

# Los dos totales corrientes se declaran por su CÓDIGO oficial, no por su
# descripción. Observe en el catálogo de arriba por qué filtrar por texto falla.
TOTAL_ACTIVO_CORRIENTE = "1D01ST"
TOTAL_PASIVO_CORRIENTE = "1D03ST"

print("Códigos declarados:", TOTAL_ACTIVO_CORRIENTE, TOTAL_PASIVO_CORRIENTE)


Empresa de referencia: AGROINDUSTRIAL LAREDO S.A.A.


,cuenta,descripcion_cuenta,monto1
0,1D0122,Activos Corrientes Distintos al Efectivo Pigno...,0.0
1,1D0118,Total Activos Corrientes Distintos de los Acti...,137966.0
2,1D0115,Activos no Corrientes o Grupos de Activos para...,0.0
3,1D01ST,Total Activos Corrientes,137966.0
4,1D0223,"Activos por Impuestos Corrientes, no Corrientes",0.0
5,1D0224,Activos no Corrientes Distintos al Efectivo Pi...,0.0
6,1D02ST,Total Activos No Corrientes,802358.0
7,1D0315,Total de Pasivos Corrientes Distintos de Pasiv...,125303.0
8,1D03ST,Total Pasivos Corrientes,125303.0
9,1D0412,"Pasivos por Impuestos Corrientes, no Corriente",0.0


Códigos declarados: 1D01ST 1D03ST


## Paso 4: Construcción reproducible de los indicadores financieros

| Familia | Indicador | Fórmula |
|---|---|---|
| Rentabilidad | Margen neto | Utilidad neta / Ingresos |
| Rentabilidad | ROA | Utilidad neta / Activo total |
| Rentabilidad | ROE | Utilidad neta / Patrimonio |
| Liquidez | Razón corriente | Activo corriente / Pasivo corriente |
| Endeudamiento | Endeudamiento | Pasivo total / Activo total |
| Endeudamiento | Apalancamiento | Pasivo total / Patrimonio |

La razón corriente se obtiene del Estado de Situación Financiera mediante las descripciones oficiales de activos y pasivos corrientes.


In [ ]:
def renombrar_primera(df, destino, candidatos):
    if destino in df.columns:
        return df
    return df.rename(columns={resolver_columna(df, candidatos): destino})

# Estandarización de las cuentas principales que devuelve obtener_InfoFinanciera.
for destino, candidatos in {
    "ventas": ["total_ingreso", "total_ingresos", "ingresos", "ventas_netas"],
    "utilidad_neta": ["utilidad_neta", "ganancia_perdida_neta", "resultado_neto"],
    "activo_total": ["activo_total", "total_activo", "total_activos"],
    "pasivo_total": ["pasivo_total", "total_pasivo", "total_pasivos"],
    "patrimonio": ["patrimonio_total", "total_patrimonio", "patrimonio"],
}.items():
    principales = renombrar_primera(principales, destino, candidatos)

columnas_base = [
    "nombre_empresa", "ventas", "utilidad_neta", "activo_total",
    "pasivo_total", "patrimonio"
]
perfil = principales[columnas_base].drop_duplicates("nombre_empresa").copy()
perfil = perfil.rename(columns={"nombre_empresa": "empresa"})

def a_numero(valor):
    '''Convierte importes de la SMV aunque incluyan separadores, espacios o paréntesis.'''
    if pd.isna(valor):
        return np.nan
    if isinstance(valor, (int, float, np.number)):
        return float(valor)
    texto = str(valor).strip().replace("\u00a0", "").replace(" ", "")
    negativo = texto.startswith("(") and texto.endswith(")")
    texto = texto.strip("()")
    # La SMV normalmente usa punto decimal. Las comas se eliminan como separador de miles.
    texto = texto.replace(",", "")
    texto = re.sub(r"[^0-9.\-]", "", texto)
    if texto in {"", "-", "."}:
        return np.nan
    numero = float(texto)
    return -numero if negativo else numero

for columna in ["ventas", "utilidad_neta", "activo_total", "pasivo_total", "patrimonio"]:
    perfil[columna] = perfil[columna].map(a_numero)

# Las cuentas se identifican por su CÓDIGO oficial y no por su descripción.
# El código es estable entre ejercicios y evita capturar cuentas cuyo texto
# también contiene "Activos Corrientes" sin ser el total: por ejemplo 1D0122,
# "Activos Corrientes Distintos al Efectivo Pignorados como Garantía Colateral",
# que vale 0 y, de usarse, produce una razón corriente igual a cero.
def extraer_total_corriente(df, codigo, nombre):
    col_codigo = resolver_columna(df, ["cuenta", "codigo_cuenta", "codigo"])
    monto = resolver_columna(df, ["monto1", "monto", "importe", "valor"])
    datos = df[df[col_codigo].astype(str).str.strip() == codigo][
        ["nombre_empresa", monto]].copy()
    datos[monto] = datos[monto].map(a_numero)
    datos = datos.dropna().drop_duplicates("nombre_empresa")
    if len(datos) < len(AZUCARERAS):
        raise ValueError(
            f"La cuenta {codigo} no está disponible para las cuatro empresas "
            f"en el ejercicio {EJERCICIO}."
        )
    return datos.rename(columns={"nombre_empresa": "empresa", monto: nombre})

perfil = perfil.merge(
    extraer_total_corriente(balance, TOTAL_ACTIVO_CORRIENTE, "activo_corriente"),
    on="empresa", how="inner")
perfil = perfil.merge(
    extraer_total_corriente(balance, TOTAL_PASIVO_CORRIENTE, "pasivo_corriente"),
    on="empresa", how="inner")

if len(perfil) < 4:
    raise ValueError("No fue posible construir el perfil completo de las cuatro empresas.")

columnas_calculo = [
    "ventas", "utilidad_neta", "activo_total", "pasivo_total", "patrimonio",
    "activo_corriente", "pasivo_corriente"
]
faltantes = perfil[columnas_calculo].isna().sum()
if faltantes.any():
    raise ValueError(
        "Existen cuentas sin valor numérico: " +
        ", ".join(f"{c} ({n})" for c, n in faltantes.items() if n)
    )

if (perfil[["ventas", "activo_total", "patrimonio", "pasivo_corriente"]] == 0).any().any():
    raise ValueError("Existe un denominador igual a cero; revise las cuentas antes de calcular.")

perfil["margen_neto"] = perfil["utilidad_neta"] / perfil["ventas"]
perfil["roa"] = perfil["utilidad_neta"] / perfil["activo_total"]
perfil["roe"] = perfil["utilidad_neta"] / perfil["patrimonio"]
perfil["razon_corriente"] = perfil["activo_corriente"] / perfil["pasivo_corriente"]
perfil["endeudamiento"] = perfil["pasivo_total"] / perfil["activo_total"]
perfil["apalancamiento"] = perfil["pasivo_total"] / perfil["patrimonio"]

numeros = perfil.select_dtypes(include="number").to_numpy()
if np.isinf(numeros[~np.isnan(numeros)]).any():
    raise ValueError("Se obtuvo una división entre cero; revise las cuentas de la empresa afectada.")

display(perfil.round(3))


,empresa,ventas,utilidad_neta,activo_total,pasivo_total,patrimonio,activo_corriente,pasivo_corriente,margen_neto,roa,roe,razon_corriente,endeudamiento,apalancamiento
0,AGROINDUSTRIAL LAREDO S.A.A.,438934.0,55453.0,940324.0,327342.0,612982.0,137966.0,125303.0,0.126,0.059,0.090,1.101,0.348,0.534
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,535074.0,91846.0,926161.0,542687.0,383474.0,294933.0,307537.0,0.172,0.099,0.240,0.959,0.586,1.415
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,637681.0,73490.0,1923732.0,874083.0,1049649.0,429304.0,362479.0,0.115,0.038,0.070,1.184,0.454,0.833
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,234045.0,9636.0,1352469.0,679594.0,672875.0,168835.0,108618.0,0.041,0.007,0.014,1.554,0.502,1.010


**Pregunta 1.** Identifique la empresa con mayor ROE y la empresa con mayor endeudamiento. ¿Coinciden? Interprete la relación observada sin afirmar causalidad.

**Respuesta:** La empresa con mayor ROE es Cartavio S.A.A. (24,0 %), la cual coincide con la empresa que presenta el mayor nivel de endeudamiento (58,6 %). Esta relación observada sugiere que el alto retorno sobre el patrimonio podría estar siendo amplificado por el apalancamiento financiero, ya que un mayor uso de deuda respecto al patrimonio tiende a magnificar el ROE cuando la rentabilidad de los activos supera el costo de dicho financiamiento. No obstante, esto constituye una asociación estadística y no una relación causal probada, dado que otros factores operativos, como el margen neto o la composición de los activos, también inciden simultáneamente en estos resultados.

---


### Reto 1 — Predicción antes del ranking (5 minutos)

**Sin ordenar todavía la tabla**, responda en equipo:

1. ¿Qué empresa creen que presenta el mejor perfil financiero?


2. Sustenten la predicción con **dos indicadores** como mínimo.


3. ¿Qué indicador podría hacer cambiar su conclusión?

**Predicción del equipo:**

**Justificación:**

Después de ejecutar el ranking del Paso 5, vuelva a esta respuesta y escriba: **¿se confirmó o no la predicción y por qué?**

Justificación: Antes de calcular los ratios financieros, la empresa destaca por registrar la mayor utilidad neta absoluta (91,846) y contar con un activo total (926,161) significativamente menor al de Casa Grande, lo que sugiere una mayor eficiencia para generar ganancias a partir de sus recursos.
Indicador que podría cambiar la conclusión: La razón corriente o el nivel de endeudamiento, ya que una alta rentabilidad podría estar sostenida por un apalancamiento excesivo o una falta de liquidez que comprometa la sostenibilidad operativa a corto plazo.
Análisis post-ranking: La predicción se confirmó en términos de rentabilidad, ya que Cartavio lideró el ranking con un ROE del 24.0 % y un margen neto del 17.2 %. Sin embargo, el análisis integral reveló que su perfil financiero conlleva un mayor riesgo, pues presentó la menor liquidez (razón corriente de 0.96) y el mayor endeudamiento (58.6 %), demostrando que su alta rentabilidad está fuertemente asociada a un apalancamiento financiero elevado.

## Paso 5: Comparación sectorial con Polars y DuckDB


In [ ]:
perfil_pl = pl.from_pandas(perfil)

ranking_roe = (
    perfil_pl
    .select([
        "empresa", "margen_neto", "roa", "roe", "razon_corriente",
        "endeudamiento", "apalancamiento"
    ])
    .sort("roe", descending=True)
)
ranking_roe


empresa,margen_neto,roa,roe,razon_corriente,endeudamiento,apalancamiento
str,f64,f64,f64,f64,f64,f64
"""CARTAVIO SOCIEDAD ANONIMA ABIE…",0.171651,0.099169,0.23951,0.959016,0.585953,1.415186
"""AGROINDUSTRIAL LAREDO S.A.A.""",0.126336,0.058972,0.090464,1.101059,0.348116,0.534016
"""CASA GRANDE SOCIEDAD ANONIMA A…",0.115246,0.038202,0.070014,1.184356,0.454368,0.832738
"""EMPRESA AGROINDUSTRIAL POMALCA…",0.041172,0.007125,0.014321,1.554392,0.502484,1.009986


In [ ]:
# Misma comparación del ranking anterior, ahora expresada en SQL sobre DuckDB.
# Ejecutar la misma pregunta en dos motores distintos y obtener el mismo
# resultado es una comprobación de reproducibilidad, no una coincidencia.
CONSULTA = """
SELECT empresa,
       ROUND(margen_neto * 100, 1)    AS margen_neto_pct,
       ROUND(roa * 100, 1)            AS roa_pct,
       ROUND(roe * 100, 1)            AS roe_pct,
       ROUND(razon_corriente, 2)      AS razon_corriente,
       ROUND(endeudamiento * 100, 1)  AS endeudamiento_pct,
       ROUND(apalancamiento, 2)       AS apalancamiento
FROM perfil
ORDER BY roe_pct DESC
"""

consulta = duckdb.sql(CONSULTA).df()
display(consulta)

# Comprobación automática: DuckDB y Polars deben coincidir.
if list(consulta["empresa"]) == list(ranking_roe["empresa"]):
    print("Correcto: DuckDB y Polars devuelven el mismo ordenamiento.")
else:
    print("Los ordenamientos no coinciden. Revise las columnas y el ORDER BY.")


,empresa,margen_neto_pct,roa_pct,roe_pct,razon_corriente,endeudamiento_pct,apalancamiento
0,CARTAVIO SOCIEDAD ANONIMA ABIERTA,17.2,9.9,24.0,0.96,58.6,1.42
1,AGROINDUSTRIAL LAREDO S.A.A.,12.6,5.9,9.0,1.10,34.8,0.53
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,11.5,3.8,7.0,1.18,45.4,0.83
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,4.1,0.7,1.4,1.55,50.2,1.01


Correcto: DuckDB y Polars devuelven el mismo ordenamiento.


In [ ]:
# Rankings complementarios para comparar dimensiones distintas del perfil.
ranking_comparado = pd.DataFrame({
    "empresa": perfil["empresa"],
    "rank_roe": perfil["roe"].rank(ascending=False, method="min").astype(int),
    "rank_liquidez": perfil["razon_corriente"].rank(ascending=False, method="min").astype(int),
    # En endeudamiento, un valor menor recibe mejor posición solo como criterio descriptivo de exposición financiera.
    "rank_endeudamiento": perfil["endeudamiento"].rank(ascending=True, method="min").astype(int),
})
ranking_comparado["cambio_max_posicion"] = (
    ranking_comparado[["rank_roe", "rank_liquidez", "rank_endeudamiento"]].max(axis=1)
    - ranking_comparado[["rank_roe", "rank_liquidez", "rank_endeudamiento"]].min(axis=1)
)
display(ranking_comparado.sort_values(["rank_roe", "empresa"]))


,empresa,rank_roe,rank_liquidez,rank_endeudamiento,cambio_max_posicion
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,1,4,4,3
0,AGROINDUSTRIAL LAREDO S.A.A.,2,3,1,2
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,3,2,2,1
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,4,1,3,3


### Reto 2 — Evidencia, inferencia y conclusión (5 minutos)

Clasifique cada afirmación como **Sustentada**, **No necesariamente** o **No puede concluirse** y justifique con cifras:

| Afirmación | Clasificación | Justificación con evidencia |
|---|---|---|
| La empresa con mayor ROE es necesariamente la más competitiva. | No necesariamente | Si bien Cartavio registra el mayor ROE (24.0 %), su competitividad integral se ve restringida por un perfil financiero de alto riesgo, sustentado en la menor liquidez del sector (razón corriente de 0.96) y el mayor nivel de endeudamiento (58.6 %). |
| Una empresa con mayor endeudamiento posee necesariamente menor madurez analítica. | No puede concluirse. | La información financiera cuantifica la estructura de capital (por ejemplo, el 58.6 % de deuda de Cartavio), pero no proporciona evidencia organizacional sobre gobierno de datos, liderazgo o talento analítico, dimensiones indispensables para evaluar el modelo DELTA. |
| Obtener un ordenamiento consistente con Polars y DuckDB fortalece la reproducibilidad del análisis. | Sustentada | La ejecución del código demuestra que Polars y DuckDB retornan exactamente la misma secuencia de empresas al ordenar por ROE, lo que valida la consistencia metodológica y elimina la dependencia de un motor de cálculo único. |

**Importante:** reproducibilidad no significa causalidad ni convierte un indicador financiero en evidencia directa de madurez organizacional.


**Pregunta 2.** Compare los rankings por rentabilidad, liquidez y endeudamiento. Explique qué revela el cambio de posición sobre el perfil competitivo de cada empresa.

**Respuesta:** La comparación de los rankings evidencia perfiles competitivos diferenciados que reflejan estrategias financieras opuestas. Cartavio lidera la rentabilidad (ROE del 24,0 %) pero ocupa el último lugar en liquidez (0,96) y endeudamiento (58,6 %), lo que revela un perfil agresivo y altamente apalancado con mayor exposición al riesgo financiero. En el extremo contrario, Pomalca registra la menor rentabilidad (1,4 %) pero la mayor liquidez (1,55), reflejando una postura conservadora que prioriza la solvencia a corto plazo en detrimento del rendimiento para el accionista. Laredo y Casa Grande mantienen posiciones intermedias; Casa Grande destaca por su perfil más homogéneo y equilibrado (con un cambio máximo de posición de 1), mientras que Laredo se distingue por ser la empresa menos endeudada (34,8 %). Estos desplazamientos demuestran que ninguna empresa lidera todas las dimensiones simultáneamente, evidenciando compensaciones directas entre la generación de valor, la liquidez y la estructura de capital.


---


## Paso 6: Visualización del perfil comparado


In [ ]:
indicadores_pct = ["margen_neto", "roa", "roe", "endeudamiento"]
nombres_indicadores = {
    "margen_neto": "Margen neto",
    "roa": "ROA",
    "roe": "ROE",
    "endeudamiento": "Endeudamiento",
}
perfil_grafico = perfil.copy()
perfil_grafico["empresa_corta"] = (
    perfil_grafico["empresa"].str.upper()
    .map(lambda x: "Laredo" if "LAREDO" in x else
                   "Cartavio" if "CARTAVIO" in x else
                   "Casa Grande" if "CASA GRANDE" in x else
                   "Pomalca" if "POMALCA" in x else x.title())
)
fig = go.Figure()
for indicador in indicadores_pct:
    fig.add_trace(go.Bar(
        name=nombres_indicadores[indicador],
        x=perfil_grafico["empresa_corta"],
        y=perfil_grafico[indicador] * 100,
        text=(perfil_grafico[indicador] * 100).round(1),
        texttemplate="%{text:.1f}%",
        textposition="outside",
        customdata=perfil_grafico[["empresa"]],
        hovertemplate="%{customdata[0]}<br>%{fullData.name}: %{y:.2f}%<extra></extra>",
    ))
fig.update_layout(
    title=f"Perfil financiero comparado — sector azucarero ({EJERCICIO})",
    barmode="group",
    yaxis_title="Porcentaje (%)",
    height=540,
    width=1050,
    template="plotly_white",
    margin=dict(t=110, b=90, l=80, r=40),
    legend=dict(orientation="h", yanchor="bottom", y=1.02, xanchor="left", x=0),
)
fig.show()

fig_liquidez = go.Figure(go.Bar(
    x=perfil_grafico["empresa_corta"],
    y=perfil_grafico["razon_corriente"],
    marker_color="#00A9CE",
    text=perfil_grafico["razon_corriente"].round(2),
    texttemplate="%{text:.2f}",
    textposition="outside",
    customdata=perfil_grafico[["empresa"]],
    hovertemplate="%{customdata[0]}<br>Razón corriente: %{y:.2f}<extra></extra>",
))
fig_liquidez.update_layout(
    title="Razón corriente por empresa",
    yaxis_title="Veces",
    height=460,
    width=900,
    template="plotly_white",
    margin=dict(t=80, b=80, l=80, r=40),
)
fig_liquidez.update_yaxes(rangemode="tozero")
fig_liquidez.show()


## Paso 7: Contexto sectorial con datos específicos del BCRP

Se emplean series oficiales sobre tipo de cambio, producción de caña y exportaciones de azúcar. Estas series describen el entorno; por sí solas no demuestran la causa de las diferencias financieras entre empresas.


In [ ]:
SERIES = {
    "PN01209PM": "tipo_cambio",
    "PN01790AM": "produccion_cana_azucar_miles_t",
    "PN38774BM": "export_azucar_fob_millones_usd",
    "PN38775BM": "volumen_exportado_azucar",
}

contexto = BCRP.get_data(SERIES, "2020-01", "2024-12")
print("Datos del BCRP:", contexto.shape)
display(contexto.tail(12))


Datos del BCRP: (60, 4)


,tipo_cambio,produccion_cana_azucar_miles_t,export_azucar_fob_millones_usd,volumen_exportado_azucar
2024-01-01,3.743667,755.602441,7.547009,10.321329
2024-02-01,3.830905,814.525062,4.398561,5.993531
2024-03-01,3.712947,563.626363,1.911468,2.355260
2024-04-01,3.718409,634.675782,3.839971,4.951844
2024-05-01,3.735273,812.740966,3.235732,3.843619
2024-06-01,3.789526,830.637431,0.481814,0.618096
2024-07-01,3.769500,840.996455,0.919740,1.246593
2024-08-01,3.745550,1017.287110,1.010892,1.358606
2024-09-01,3.772143,930.472111,3.097093,4.088378
2024-10-01,3.757500,1053.902968,1.906427,2.543543


In [ ]:
# Evolución temporal del contexto sectorial. Cada serie se normaliza a índice 100
# para comparar tendencias con unidades diferentes, sin alterar los datos originales.
contexto_grafico = contexto.copy()
columnas_numericas = contexto_grafico.select_dtypes(include="number").columns
base = contexto_grafico[columnas_numericas].apply(lambda s: s.dropna().iloc[0])
indice_100 = contexto_grafico[columnas_numericas].divide(base).multiply(100)

fig_contexto = go.Figure()
for columna in indice_100.columns:
    fig_contexto.add_trace(go.Scatter(
        x=indice_100.index,
        y=indice_100[columna],
        mode="lines",
        name=columna,
    ))
fig_contexto.update_layout(
    title="Contexto del sector azucarero — índice base 100",
    yaxis_title="Índice",
    xaxis_title="Periodo",
    height=480,
)
fig_contexto.show()


**Pregunta 3.** Describa la evolución del tipo de cambio, la producción de caña y las exportaciones de azúcar. Relacione ese contexto con el perfil financiero y diferencie expresamente:

- **Evidencia:** resultado directamente observado.
- **Inferencia:** interpretación razonada basada en la evidencia.
- **Opinión:** valoración que debe justificarse.

**Respuesta:**  Las series oficiales del BCRP muestran un comportamiento sectorial generalmente favorable, con incrementos en la producción de caña y en el volumen de exportaciones de azúcar, junto con un tipo de cambio que se mantiene en un rango que beneficia a las empresas exportadoras. Sin embargo, al relacionar este contexto con el perfil financiero, se observa que un entorno externo positivo no garantiza resultados homogéneos: mientras Cartavio logra un ROE del 24,0 %, Pomalca registra apenas un 1,4 %, lo que indica que los factores internos de gestión predominan sobre las condiciones generales del mercado.

* Evidencia: Las series del BCRP registran un crecimiento en la producción de caña y en las exportaciones de azúcar durante el periodo analizado, y los estados financieros muestran un ROE del 24,0 % para Cartavio y del 1,4 % para Pomalca.
* Inferencia: Las marcadas diferencias en rentabilidad y endeudamiento entre las empresas, a pesar de operar bajo un mismo entorno sectorial favorable, indican que la eficiencia operativa, el control de costos y la estructura financiera interna son los principales determinantes de su desempeño individual.
* Opinión: La gerencia de Pomalca debería priorizar una revisión urgente de su estructura de costos y apalancamiento, ya que su baja rentabilidad no puede atribuirse a una contracción del mercado, sino a decisiones internas que limitan su competitividad frente a sus pares.

---


### Reto 3 — Diagnóstico provisional de madurez y DELTA (8 minutos)

Seleccione **una** de las cuatro empresas. No se busca “adivinar” su madurez: se busca reconocer qué puede y qué no puede sostenerse con la evidencia disponible.

| Aspecto | Respuesta del equipo |
|---|---|
| Empresa seleccionada | Cartavio S.A.A. |
| Evidencia financiera observada | Mayor ROE (24.0 %), mayor endeudamiento (58.6 %) y menor liquidez (razón corriente de 0.96) en el ejercicio 2024. |
| Estadio de evolución analítica compatible **de manera provisional** | No es asignable, ya que el desempeño financiero no constituye evidencia de procesos, cultura ni uso transversal de la analítica en la organización. |
| Dimensión DELTA evaluable parcialmente | Data (Datos), únicamente en lo referente a su capacidad para generar, estructurar y reportar información financiera auditada ante la SMV. |
| Dimensiones DELTA que no pueden verificarse | Enterprise, Leadership, Targets, Analysts, Technology y Analytical Techniques, pues requieren evidencia de procesos internos, talento y gobernanza. |
| Evidencia organizacional adicional requerida | Arquitectura de datos interna, existencia de un área de analítica, métricas operativas y registros de decisiones estratégicas documentadas basadas en evidencia. |

**Cierre del reto:** redacte una frase que distinga claramente **evidencia**, **inferencia** y **opinión**.


### Pausa metodológica — ¿qué NO podemos afirmar?

Los estados financieros permiten comparar resultados y estructura financiera, pero **no prueban por sí solos** liderazgo analítico, cultura de datos, disponibilidad de analistas, integración empresarial de la analítica ni uso de IA generativa. Por ello, cualquier estadio o perfil DELTA derivado de este laboratorio debe presentarse como **hipótesis provisional**, no como diagnóstico organizacional definitivo.


## Actividad 3 — Informe de perfil analítico comparado

Responda en términos de negocio y sustente cada afirmación con las cifras obtenidas.

**A.** Interprete tres hallazgos del perfil comparado y explique, para cada uno, su implicancia competitiva en el sector.

**Respuesta:** El primer hallazgo es que Cartavio lidera la rentabilidad (ROE del 24,0 % y margen del 17,2 %) pero presenta la menor liquidez (0,96) y mayor endeudamiento (58,6 %); su implicancia competitiva es que su ventaja en generación de valor depende de un apalancamiento agresivo que eleva su vulnerabilidad ante shocks de liquidez. El segundo hallazgo es que Pomalca registra la menor rentabilidad (ROE del 1,4 % y margen del 4,1 %) pese a tener la mayor liquidez (1,55); esto implica una ineficiencia operativa donde la solvencia a corto plazo no se está traduciendo en competitividad ni en retorno para el accionista. Finalmente, Laredo y Casa Grande muestran perfiles intermedios y equilibrados (como el 34,8 % de endeudamiento de Laredo); su implicancia es que priorizan la estabilidad estructural, lo que les otorga resiliencia operativa, aunque limita su capacidad de expansión financiera frente a la agresividad de Cartavio.

**B.** Determine si los estados financieros permiten ubicar de manera concluyente a cada empresa en uno de los cinco estadios de evolución analítica. Formule, cuando corresponda, una **clasificación provisional** y precise qué evidencia organizacional adicional se necesita para confirmarla.

**Respuesta:** Los estados financieros no permiten ubicar de manera concluyente a las empresas en un estadio de evolución analítica, ya que miden resultados económicos históricos y no capacidades organizacionales transversales. De forma provisional, podrían clasificarse en el estadio 2 (Analítica localizada) o 3 (Aspiración analítica), dado que demuestran capacidad para procesar y reportar información regulatoria. Para confirmar su estadio real, se requiere evidencia organizacional adicional, como la existencia de un gobierno de datos centralizado, la integración de métricas operativas en la toma de decisiones directivas y el uso de modelos predictivos o prescriptivos más allá del análisis financiero tradicional.

**C.** A partir del modelo DELTA, ¿qué dimensión resulta más difícil de evaluar únicamente con estados financieros? Explique por qué y qué información adicional se requeriría.

**Respuesta:** Las dimensiones de Liderazgo y Analistas son las más difíciles de evaluar únicamente con estados financieros. Los indicadores financieros reflejan el resultado final de las decisiones, pero no revelan si la alta dirección exige sistemáticamente evidencia empírica para decidir (Liderazgo) ni si la organización cuenta con talento especializado que conecte los datos con la estrategia del negocio (Analistas). Para evaluarlas, se necesitaría evidencia interna, como actas de directorio donde se documenten decisiones estratégicas basadas en datos, organigramas que reflejen la integración de roles analíticos en las áreas de negocio y la existencia de una cultura de medición compartida.

**D.** Formule dos recomendaciones sustentadas exclusivamente en los datos observados, dirigidas a la empresa con el perfil competitivo más débil.

**Respuesta:** Dirigidas a Pomalca, que presenta el perfil competitivo más débil en rentabilidad (ROE del 1,4 % y margen del 4,1 %):

1. Optimizar la gestión de capital de trabajo, ya que su alta razón corriente (1,55) combinada con una baja rentabilidad sugiere la existencia de recursos ociosos que deberían reorientarse hacia la reducción de su elevado endeudamiento (50,2 %) para disminuir costos financieros.
2. Implementar un programa de control de costos y eficiencia industrial, pues su margen neto es el más bajo del sector a pesar de operar en un contexto macroeconómico favorable, evidenciado por el crecimiento en las exportaciones y la producción de caña de azúcar reportadas por el BCRP.

**E.** Explique cómo la inteligencia artificial generativa podría modificar el análisis de competitividad. Distinga entre usos plausibles y afirmaciones no verificables.

**Respuesta:** La inteligencia artificial generativa podría modificar el análisis acelerando las fases descriptiva y diagnóstica, por ejemplo, automatizando la detección de anomalías en los datos operativos o generando resúmenes ejecutivos de las variaciones financieras (uso plausible). Sin embargo, es una afirmación no verificable sostener que la IA por sí sola puede definir la estrategia competitiva o garantizar ventajas sostenibles; al actuar como un amplificador, si los datos subyacentes son fragmentados o los objetivos estratégicos son ambiguos, la IA solo acelerará la generación de conclusiones erróneas sin resolver las deficiencias estructurales de la organización.


### Matriz de síntesis para la decisión

Complete esta matriz antes de redactar las conclusiones finales:

| Empresa | Fortaleza principal | Riesgo/debilidad principal | Evidencia numérica | Implicancia competitiva |
|---|---|---|---|---|
| Casa Grande | Mayor escala operativa y solvencia a corto plazo equilibrada. | Baja rentabilidad sobre activos y patrimonio respecto a su gran base de activos. | Mayor activo total (1 923 732) y ventas (637 681), pero ROE de 7,0 % y ROA de 3,8 %. | Ineficiencia en el uso de su extensa base de activos para generar retornos proporcionales a su tamaño. |
| Cartavio | Máxima rentabilidad y eficiencia en la generación de utilidades. | Alta exposición financiera y tensión de liquidez a corto plazo. | Mayor ROE (24,0 %), margen neto (17,2 %) y ROA (9,9 %), pero menor razón corriente (0,96) y mayor endeudamiento (58,6 %). | Su ventaja competitiva en retorno depende de un apalancamiento agresivo que la hace vulnerable a shocks de liquidez. |
| Agroindustrial Laredo | Estructura financiera conservadora y equilibrada con rentabilidad moderada. | Menor escala de ventas y retorno absoluto comparado con el líder del sector. | Menor endeudamiento (34,8 %), razón corriente adecuada (1,10), ROE de 9,0 % y margen de 12,6 %. | Alta resiliencia operativa, aunque con menor capacidad de expansión financiera frente a competidores más agresivos. |
| Pomalca | Alta liquidez para cubrir obligaciones de corto plazo. | Baja rentabilidad general y alta dependencia de deuda pese a su liquidez. | Mayor razón corriente (1,55), pero menor margen neto (4,1 %), ROA (0,7 %) y ROE (1,4 %), con un endeudamiento del 50,2 %. | Ineficiencia operativa severa; los recursos líquidos no se están traduciendo en rentabilidad ni en ventaja competitiva. |

Luego responda: **¿qué empresa requiere mayor atención gerencial según los indicadores observados y qué dos acciones recomendaría sin exceder la evidencia disponible?**


---

## Reto de aplicación y retroalimentación

En esta sección se aplicarán los procedimientos desarrollados durante la sesión a nuevas situaciones de análisis. Cada ejercicio requiere modificar, completar o construir código a partir de los datos financieros previamente procesados. Posteriormente, los resultados obtenidos deberán interpretarse brevemente desde una perspectiva empresarial. El propósito es comprobar la comprensión de las técnicas utilizadas y fortalecer la capacidad de adaptar el análisis ante nuevas preguntas de negocio.

**Indicaciones generales.** Los cinco ejercicios operan sobre `perfil_reto`, copia de trabajo del perfil financiero construido en la Actividad 2. Se emplea una copia para que ninguna modificación introducida en esta sección altere los resultados obtenidos con anterioridad. Los datos proceden íntegramente de la SMV; no corresponde generar ni sustituir valores en ningún caso. Cada respuesta escrita no debe exceder cuatro líneas.

**Duración en sesión:** 25 minutos. Los ejercicios que no concluyan durante la sesión se completan como avance del entregable y se depositan en el repositorio del equipo junto con el notebook ejecutado.


In [ ]:
# Copia de trabajo para la sección de retos.
perfil_reto = perfil.copy()
print("Copia de trabajo creada:", perfil_reto.shape)
display(perfil_reto[["empresa", "ventas", "activo_total", "margen_neto", "roa", "roe"]].round(3))


Copia de trabajo creada: (4, 14)


,empresa,ventas,activo_total,margen_neto,roa,roe
0,AGROINDUSTRIAL LAREDO S.A.A.,438934.0,940324.0,0.126,0.059,0.090
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,535074.0,926161.0,0.172,0.099,0.240
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,637681.0,1923732.0,0.115,0.038,0.070
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,234045.0,1352469.0,0.041,0.007,0.014


### Ejercicio 1 — Recuperación de una nueva cuenta financiera del catálogo oficial

Durante la Actividad 2 se identificaron dos cuentas del Estado de Situación Financiera a partir de su **código oficial** y no de su descripción textual. Ese procedimiento constituye el criterio de trazabilidad del laboratorio, dado que el código permanece estable entre ejercicios mientras que la descripción admite variantes que conducen a errores silenciosos.

En este ejercicio deberá replicar el mismo procedimiento sobre una cuenta que aún no ha sido utilizada: el **Total de Activos No Corrientes**. Esta cuenta reúne los recursos de largo plazo de la empresa —terrenos, plantaciones, maquinaria e instalaciones industriales— y su participación sobre el activo total describe la estructura de inversión de cada empresa azucarera.

**Se solicita:**

1. Localizar en el catálogo de la SMV la cuenta correspondiente al Total de Activos No Corrientes e identificar su código oficial.
2. Recuperar el valor de esa cuenta para las cuatro empresas analizadas.
3. Incorporar la variable a `perfil_reto`.
4. Calcular el porcentaje que dicha cuenta representa respecto del activo total.

**Tiempo estimado:** 10 a 12 minutos.


In [ ]:
# ---------------------------------------------------------------------------
# CELDA DE TRABAJO — Ejercicio 1
# Desarrolle aquí su código.
# ---------------------------------------------------------------------------

# 1. Código oficial identificado en el catálogo del Paso 3
# (Cuenta: "Total Activos No Corrientes")
TOTAL_ACTIVO_NO_CORRIENTE = "1D02ST"

# 2 y 3. Recuperar el valor para las cuatro empresas e incorporar a perfil_reto
perfil_reto = perfil_reto.merge(
    extraer_total_corriente(balance, TOTAL_ACTIVO_NO_CORRIENTE, "activo_no_corriente"),
    on="empresa", how="inner"
)

# 4. Calcular el porcentaje que representa respecto al activo total
perfil_reto["pct_activo_no_corriente"] = (
    perfil_reto["activo_no_corriente"] / perfil_reto["activo_total"]
)

# Visualización del resultado
display(
    perfil_reto[["empresa", "activo_no_corriente", "activo_total", "pct_activo_no_corriente"]]
    .round(3)
)


,empresa,activo_no_corriente,activo_total,pct_activo_no_corriente
0,AGROINDUSTRIAL LAREDO S.A.A.,802358.0,940324.0,0.853
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,631228.0,926161.0,0.682
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,1494428.0,1923732.0,0.777
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,1183634.0,1352469.0,0.875


**Pregunta 4.** ¿Qué empresa presenta la mayor proporción de activos no corrientes y qué podría indicar esto sobre su estructura de inversión?

**Respuesta (máximo cuatro líneas):** Empresa Agroindustrial Pomalca S.A.A. presenta la mayor proporción de activos no corrientes (87,5 %). Esto indica una estructura de inversión altamente intensiva en capital, con la mayor parte de sus recursos inmovilizados en activos de largo plazo. Dicha configuración sugiere una menor flexibilidad operativa y una posible restricción en su liquidez a corto plazo frente a sus competidores.


---


### Ejercicio 2 — Construcción de un indicador de eficiencia en el uso de los activos

Los indicadores calculados en la Actividad 2 describen rentabilidad, liquidez y endeudamiento, pero ninguno mide la eficiencia con que la empresa emplea su base de activos para generar ventas. La rotación de activos cubre esa dimensión: expresa cuántas unidades monetarias de venta produce cada unidad monetaria invertida en activos, y se obtiene dividiendo las ventas entre el activo total.

En el sector azucarero este indicador resulta particularmente informativo, por tratarse de empresas intensivas en capital, con extensas superficies de cultivo e instalaciones industriales. Dos empresas pueden alcanzar rentabilidades semejantes mediante estrategias opuestas: una sostenida en márgenes elevados y otra en un mayor volumen de operación sobre la misma base de activos.

**Se solicita:**

1. Calcular la rotación de activos e incorporarla a `perfil_reto` en la columna `rotacion_activos`.
2. Generar un ranking de las empresas ordenado de mayor a menor según dicho indicador.
3. Comparar ese ordenamiento con el obtenido a partir del ROA calculado en la Actividad 2.

**Tiempo estimado:** 10 a 12 minutos.


In [ ]:
# ---------------------------------------------------------------------------
# CELDA DE TRABAJO — Ejercicio 2
# Desarrolle aquí su código.
# ---------------------------------------------------------------------------

# 1. Calcular la rotación de activos e incorporarla a perfil_reto
perfil_reto["rotacion_activos"] = perfil_reto["ventas"] / perfil_reto["activo_total"]

# 2. Generar un ranking de las empresas ordenado de mayor a menor según dicho indicador.
# Se incluye la columna 'roa' para facilitar la comparación solicitada en el punto 3.
ranking_rotacion = perfil_reto[["empresa", "rotacion_activos", "roa"]].sort_values(
    by="rotacion_activos", ascending=False
).reset_index(drop=True)

# 3. Visualización del resultado para comparar el ordenamiento
display(ranking_rotacion.round(4))

,empresa,rotacion_activos,roa
0,CARTAVIO SOCIEDAD ANONIMA ABIERTA,0.5777,0.0992
1,AGROINDUSTRIAL LAREDO S.A.A.,0.4668,0.0590
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,0.3315,0.0382
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,0.1731,0.0071


**Pregunta 5.** ¿La empresa con mayor rotación de activos presenta también el mayor ROA? Explique brevemente a partir de los resultados obtenidos.

**Respuesta (máximo cuatro líneas):** Sí, Cartavio presenta tanto la mayor rotación de activos (0,578) como el mayor ROA (9,9 %). Esto evidencia que su liderazgo en rentabilidad sobre los activos está impulsado simultáneamente por una mayor eficiencia operativa para generar ventas a partir de su base instalada y por su superior margen neto frente al resto del sector.

---


### Ejercicio 3 — Verificación numérica de la descomposición DuPont del ROA

La descomposición DuPont establece que la rentabilidad sobre activos puede expresarse como el producto de dos componentes: el margen neto, que mide la rentabilidad obtenida por cada unidad vendida, y la rotación de activos, que mide la intensidad con que se emplea la base de activos.

El valor analítico de esta descomposición no reside en la identidad algebraica, sino en la explicación que aporta: permite establecer si una rentabilidad elevada proviene del precio y del control de costos, del volumen de operación, o de la concurrencia de ambos factores. Corresponde comprobar numéricamente la identidad sobre los datos reales de la SMV, dado que toda relación teórica empleada como evidencia debe verificarse antes de sustentar una conclusión.

**Se solicita:**

1. Construir la columna `roa_dupont` como el producto del margen neto por la rotación de activos.
2. Construir la columna `diferencia_roa` como la diferencia entre el ROA obtenido en la Actividad 2 y `roa_dupont`.
3. Verificar numéricamente la magnitud de esa diferencia y determinar si resulta admisible, considerando el redondeo de los importes reportados por la SMV.

**Tiempo estimado:** 10 a 12 minutos.


In [ ]:
# ---------------------------------------------------------------------------
# CELDA DE TRABAJO — Ejercicio 3
# Desarrolle aquí su código.
# ---------------------------------------------------------------------------

# 1. Construir la columna roa_dupont como el producto del margen neto por la rotación de activos
perfil_reto["roa_dupont"] = perfil_reto["margen_neto"] * perfil_reto["rotacion_activos"]

# 2. Construir la columna diferencia_roa como la diferencia entre el ROA de la Actividad 2 y roa_dupont
perfil_reto["diferencia_roa"] = perfil_reto["roa"] - perfil_reto["roa_dupont"]

# 3. Verificar numéricamente la magnitud de esa diferencia
# Se muestran las columnas relevantes con alta precisión decimal para inspeccionar la diferencia real
display(
    perfil_reto[["empresa", "roa", "roa_dupont", "diferencia_roa"]]
    .round(8)
)

# Cálculo de la magnitud máxima de la diferencia para validar su admisibilidad
max_diferencia = perfil_reto["diferencia_roa"].abs().max()
print(f"\nMagnitud máxima de la diferencia absoluta: {max_diferencia:.2e}")

if max_diferencia < 1e-10:
    print("Conclusión: La diferencia es despreciable (del orden de 1e-16). Esto es totalmente admisible y se debe únicamente a limitaciones de precisión en la aritmética de punto flotante de Python, no a un error en los datos o en la fórmula.")
else:
    print("Conclusión: Existe una diferencia significativa que requiere revisión de las fórmulas o de los datos originales.")

,empresa,roa,roa_dupont,diferencia_roa
0,AGROINDUSTRIAL LAREDO S.A.A.,0.058972,0.058972,-0.0
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,0.099169,0.099169,0.0
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,0.038202,0.038202,0.0
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,0.007125,0.007125,0.0



Magnitud máxima de la diferencia absoluta: 6.94e-18
Conclusión: La diferencia es despreciable (del orden de 1e-16). Esto es totalmente admisible y se debe únicamente a limitaciones de precisión en la aritmética de punto flotante de Python, no a un error en los datos o en la fórmula.


**Pregunta 6.** ¿Qué componente parece explicar mejor el ROA de la empresa líder: el margen neto, la rotación de activos o ambos?

**Respuesta (máximo cuatro líneas):** Aunque ambos componentes actúan de forma concurrente, el margen neto (17,2 %) explica de manera más determinante el liderazgo de Cartavio, ya que su ventaja respecto al resto del sector es mucho más pronunciada que la de su rotación de activos (0,58). Esto indica que su superior rentabilidad sobre los activos se sustenta principalmente en una mayor eficiencia en costos y precios, más que en un volumen de ventas desproporcionado sobre su base instalada.


---


### Ejercicio 4 — Consulta empresarial con condiciones múltiples en DuckDB

En la Actividad 2 se empleó DuckDB para reproducir un ordenamiento ya obtenido con Polars, con la finalidad de comprobar la reproducibilidad del resultado entre dos motores distintos. Una consulta de negocio, sin embargo, no se limita a ordenar: delimita un subconjunto de la población mediante criterios definidos previamente por la gerencia.

En este ejercicio deberá construir una consulta que identifique aquellas empresas que satisfacen **simultáneamente** dos condiciones financieras —capacidad de atender obligaciones de corto plazo y generación de rentabilidad positiva para el accionista— y que presente el resultado ordenado según su exposición al endeudamiento. Este es el procedimiento habitual mediante el cual un área de riesgos delimita una cartera de empresas elegibles.

**Se solicita:**

1. Construir sobre `perfil_reto` una consulta SQL que incluya `SELECT`, `WHERE`, condiciones múltiples enlazadas con `AND` y `ORDER BY`.
2. Establecer como criterios una razón corriente mayor o igual a 1 y un ROE positivo.
3. Ordenar el resultado por endeudamiento, de menor a mayor.
4. Determinar cuántas empresas satisfacen ambos criterios.

**Tiempo estimado:** 10 a 12 minutos.


In [ ]:
# ---------------------------------------------------------------------------
# CELDA DE TRABAJO — Ejercicio 4
# Desarrolle aquí su código.
# ---------------------------------------------------------------------------

import duckdb

# 1. Construir la consulta SQL sobre perfil_reto con las condiciones solicitadas
consulta_sql = """
    SELECT empresa, razon_corriente, roe, endeudamiento
    FROM perfil_reto
    WHERE razon_corriente >= 1 AND roe > 0
    ORDER BY endeudamiento ASC
"""

# 2. Ejecutar la consulta y visualizar el subconjunto de empresas elegibles
resultado = duckdb.sql(consulta_sql).df()
display(resultado)

# 3. Determinar cuántas empresas satisfacen ambos criterios
cantidad_empresas = len(resultado)
print(f"\nCantidad de empresas que satisfacen ambos criterios: {cantidad_empresas}")

,empresa,razon_corriente,roe,endeudamiento
0,AGROINDUSTRIAL LAREDO S.A.A.,1.101059,0.090464,0.348116
1,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,1.184356,0.070014,0.454368
2,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,1.554392,0.014321,0.502484



Cantidad de empresas que satisfacen ambos criterios: 3


**Pregunta 7.** Según los criterios establecidos, ¿qué empresa presenta la condición financiera más favorable dentro del grupo analizado?

**Respuesta (máximo cuatro líneas):**  Agroindustrial Laredo S.A.A. presenta la condición financiera más favorable, ya que cumple con los criterios de liquidez (razón corriente de 1,10) y rentabilidad (ROE del 9,0 %), y además registra el menor nivel de endeudamiento (34,8 %) entre las empresas que satisfacen ambas condiciones, lo que indica una menor exposición al riesgo financiero.

---


### Ejercicio 5 — Representación conjunta de la rentabilidad y del nivel de endeudamiento

Los gráficos de barras construidos en la Actividad 2 comparan una dimensión por vez. La decisión empresarial, sin embargo, rara vez depende de un solo indicador: una rentabilidad elevada sostenida sobre un endeudamiento igualmente elevado no equivale a una rentabilidad semejante obtenida con una estructura financiera conservadora, aun cuando ambas presenten el mismo ROE.

El gráfico de dispersión permite examinar dos dimensiones de manera simultánea y, mediante el tamaño del marcador, incorporar una tercera. Al situar el endeudamiento en el eje horizontal y el ROE en el eje vertical, cada cuadrante adquiere una lectura gerencial precisa: la combinación de rentabilidad elevada con endeudamiento reducido corresponde al perfil más favorable, mientras que la combinación inversa exige un examen detenido.

**Se solicita:**

1. Construir con Plotly un gráfico de dispersión que sitúe el endeudamiento en el eje horizontal y el ROE en el eje vertical.
2. Identificar cada punto con el nombre de la empresa correspondiente.
3. Representar el activo total mediante el tamaño del marcador.
4. Rotular los ejes y titular el gráfico de modo que resulte interpretable sin recurrir al código.

**Tiempo estimado:** 10 a 12 minutos.


In [ ]:
# ---------------------------------------------------------------------------
# CELDA DE TRABAJO — Ejercicio 5
# Desarrolle aquí su código.
# ---------------------------------------------------------------------------

import plotly.express as px

# 1. Construir el gráfico de dispersión situando endeudamiento en X y ROE en Y
fig = px.scatter(
    perfil_reto,
    x="endeudamiento",
    y="roe",
    size="activo_total",          # 3. Representar el activo total mediante el tamaño del marcador
    hover_name="empresa",         # 2. Identificar cada punto al pasar el cursor
    text="empresa",               # 2. Mostrar el nombre de la empresa directamente en el gráfico
    title="Perfil Financiero: Rentabilidad (ROE) vs. Nivel de Endeudamiento (2024)", # 4. Titular interpretable
    labels={
        "endeudamiento": "Nivel de Endeudamiento (Pasivo Total / Activo Total)",
        "roe": "Rentabilidad sobre el Patrimonio (ROE)",
        "activo_total": "Activo Total (Tamaño del marcador)"
    }
)

# Ajustar la posición del texto para evitar superposiciones con los marcadores
fig.update_traces(textposition="top center")

# 4. Rotular y formatear los ejes como porcentajes para una lectura gerencial clara
fig.update_layout(
    xaxis=dict(tickformat=".1%"),
    yaxis=dict(tickformat=".1%"),
    height=550,
    width=850,
    template="plotly_white"
)

fig.show()

**Pregunta 8.** ¿Qué empresa muestra la combinación más favorable entre rentabilidad y nivel de endeudamiento?

**Respuesta (máximo cuatro líneas):**  Agroindustrial Laredo S.A.A. presenta la combinación más favorable, ubicándose en el cuadrante superior izquierdo del gráfico con un ROE del 9,0 % y el menor nivel de endeudamiento del sector (34,8 %). Esta posición refleja una rentabilidad sólida sostenida sobre una estructura financiera conservadora, a diferencia de Cartavio, cuyo ROE del 24,0 % depende de un apalancamiento elevado (58,6 %) que incrementa su exposición al riesgo financiero.

---


### Retroalimentación final — Vinculación con el modelo DELTA Plus

Los cinco ejercicios anteriores se resolvieron mediante Python, DuckDB y Plotly sobre datos financieros oficiales. Corresponde ahora situar ese trabajo dentro del marco conceptual de la sesión. Responda de manera breve.

**Pregunta 9.** A partir del análisis realizado mediante Python, identifique qué dimensión de DELTA Plus se evidencia directamente en esta actividad: *Technology*, *Analytical Techniques* o ambas. Justifique su respuesta en un máximo de tres líneas.

**Respuesta:**  Se evidencian ambas dimensiones. Technology se manifiesta en el uso de Python, DuckDB y Plotly para un procesamiento de datos reproducible, mientras que Analytical Techniques se refleja en la aplicación de métodos descriptivos y diagnósticos, como el cálculo de razones financieras y la descomposición DuPont, para evaluar el desempeño.


---

**Pregunta 10.** *Technology.* ¿Qué tecnologías resultarían necesarias para que una de las empresas analizadas pudiera ejecutar de manera periódica y reproducible el análisis desarrollado en este laboratorio?

**Respuesta (máximo cuatro líneas):**  Sería necesaria una arquitectura de datos que incluya un almacén centralizado (Data Warehouse o Data Lake) para la ingesta automatizada de datos de la SMV y el BCRP, herramientas de orquestación de flujos ETL (como Apache Airflow) para programar las extracciones periódicas, y una plataforma de visualización empresarial (como Power BI o Tableau) para el despliegue continuo de los indicadores.


---

**Pregunta 11.** *Analytical Techniques.* ¿En qué nivel de técnicas analíticas se sitúa el análisis realizado en esta sesión y qué técnica adicional podría incorporar la organización para mejorar sus decisiones?

**Respuesta (máximo cuatro líneas):**  El análisis se sitúa en los niveles descriptivo y diagnóstico, dado que resume el desempeño histórico y explica las causas de la rentabilidad mediante razones financieras y la descomposición DuPont. Para mejorar sus decisiones, la organización podría incorporar técnicas predictivas, como modelos de series de tiempo para pronosticar flujos de caja, o prescriptivas, como modelos de optimización de la estructura de capital.


---


## Conclusiones

Registre tres conclusiones técnicas y de negocio:

1. Las empresas del sector azucarero presentan perfiles financieros diferenciados que revelan estrategias opuestas: Cartavio lidera la rentabilidad (ROE del 24,0 %) pero con el mayor riesgo financiero (endeudamiento del 58,6 % y razón corriente de 0,96), mientras que Pomalca prioriza la liquidez (1,55) sin lograr rentabilidad competitiva (ROE del 1,4 %). Esta evidencia demuestra que no existe una empresa que lidere simultáneamente todas las dimensiones financieras, lo que obliga a evaluar el desempeño de manera integral y no mediante indicadores aislados.

2. Los estados financieros no constituyen evidencia suficiente para diagnosticar la madurez analítica de una organización: Aunque las cuatro empresas reportan información auditada ante la SMV, esto solo permite evaluar parcialmente la dimensión "Datos" del modelo DELTA. Determinar el estadio de evolución analítica requiere evidencia organizacional adicional sobre liderazgo basado en evidencia, integración transversal de la analítica, talento especializado y uso de técnicas avanzadas, aspectos que no son observables en los reportes financieros tradicionales.

3. La reproducibilidad del análisis se fortalece al emplear múltiples motores de procesamiento (Polars y DuckDB) y al basar la extracción de datos en códigos oficiales de cuenta: La verificación de que ambos motores producen el mismo ordenamiento de empresas valida la consistencia metodológica, mientras que el uso de códigos estables (como "1D01ST" para Activos Corrientes) en lugar de descripciones textuales previene errores silenciosos que podrían comprometer la calidad de la evidencia generada para la toma de decisiones.


## Referencias oficiales

- Superintendencia del Mercado de Valores. Portal de Datos Abiertos y servicio de información financiera.
- Banco Central de Reserva del Perú. Series estadísticas mensuales.
